File: padroesCriacionais/padrao-de-projeto-abstractFactory/php/tests/unit/GameTest.php
<?php

namespace Abstractfactory\tests\unit;

use Abstractfactory\Php\AbstractEasyGameFactory;
use Abstractfactory\Php\AbstractHardGameFactory;
use Abstractfactory\Php\Game;
use PHPUnit\Framework\TestCase;

class GameTest extends TestCase
{
    public function testShouldReturnEasyEnemies(): void
    {
        $easyGameFactory = new AbstractEasyGameFactory();
        $game = new Game($easyGameFactory);
        $this->assertEquals("gera inimigos no modo facil", $game->getEnemies());
    }

    public function testShouldReturnEasyMap(): void
    {
        $easyGameFactory = new AbstractEasyGameFactory();
        $game = new Game($easyGameFactory);
        $this->assertEquals("gera mapa no modo facil", $game->getMap());
    }

    public function testShouldReturnHardEnemies(): void
    {
        $hardGameFactory = new AbstractHardGameFactory();
        $game = new Game($hardGameFactory);
        $this->assertEquals("gera inimigos no modo dificil", $game->getEnemies());
    }

    public function testShouldReturnHardMap(): void
    {
        $hardGameFactory = new AbstractHardGameFactory();
        $game = new Game($hardGameFactory);
        $this->assertEquals("gera mapa no modo dificil", $game->getMap());
    }
}

File: padroesCriacionais/padrao-de-projeto-abstractFactory/java/src/test/java/GameTest.java
import abstractFactory.AbstractEasyGameFactory;
import abstractFactory.AbstractHardGameFactory;
import abstractFactory.Game;
import abstractFactory.iAbstractGameFactory;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class GameTest {

    @Test
    public void shouldReturnEasyEnemies() {
        iAbstractGameFactory easyGameFactory = new AbstractEasyGameFactory();
        Game game = new Game(easyGameFactory);
        assertEquals("gera inimigos no modo facil", game.getEnemies());
    }

    @Test
    public void shoudlReturnEasyMap() {
        iAbstractGameFactory easyGameFactory = new AbstractEasyGameFactory();
        Game game = new Game(easyGameFactory);
        assertEquals("gera mapa no modo facil", game.getMap());
    }

    @Test
    public void shouldReturnHardEnemies() {
        iAbstractGameFactory easyGameFactory = new AbstractHardGameFactory();
        Game game = new Game(easyGameFactory);
        assertEquals("gera inimigos no modo dificil", game.getEnemies());
    }

    @Test
    public void shoudlReturnHardMap() {
        iAbstractGameFactory easyGameFactory = new AbstractHardGameFactory();
        Game game = new Game(easyGameFactory);
        assertEquals("gera mapa no modo dificil", game.getMap());
    }
}
